modules/core/src/main/java/org/apache/ignite/spi/communication/tcp/internal/CommunicationTcpUtils.java

```diff
@@ -201,4 +205,161 @@ public static void failNode(ClusterNode nodeToFail,
+         * Creates a new instance of network interface matcher.
+         * @param iface Pattern of network interface.
+         */
+        public NetworkInterfaceMatcher(String iface) {
+            this.iface = iface;
+        }
+
+        /**
+         * Gets the pattern of network interface.
+         * @return Pattern of network interface.
+         */
+        public String networkInterface() {
+            return iface;
+        }
+
+        /**
+         * Matches the given address against the pattern.
+         *
+         * @param addr Address to match.
+         * @return {@code true} if the address matches the pattern, {@code false} otherwise.
+         */
+        abstract boolean matches(InetAddress addr);
+
+        /** {@inheritDoc} */
+        @Override public String toString() {
+            return S.toString(NetworkInterfaceMatcher.class, this);
+        }
+    }
+
+    /**
+     * Matcher for IPv4 addresses.
+     */
+    public static class IPv4Matcher extends NetworkInterfaceMatcher {
+        private static final Pattern DIGIT_PATTERN = Pattern.compile("\\d{1,3}");
+        private static final Pattern RANGE_PATTERN = Pattern.compile("\\d{1,3}\\s*-\\s*\\d{1,3}");
+        private static final String WILDCARD_PATTERN = "*";
+
+        /** Maximum allowed value of IPv4 octet. */
+        private static final int IPV4_MAX_OCTET_VALUE = 255;
+
+        /** Shared predicate that always returns {@code true}. */
+        private static final IgnitePredicate<String> ALWAYS_TRUE = new AlwaysTruePredicate<>();
+
+        /** List of predicates for each segment. */
+        private final IgnitePredicate<String>[] segmentPred = new IgnitePredicate[4];
+
+        /**
+         * Creates a new instance of IPv4 address matcher.
+         * @param interfacePattern Pattern of network interface.
+         */
+        public IPv4Matcher(String interfacePattern) {
+            super(interfacePattern);
+
+            String[] segments = networkInterface().split("\\.");
+            if (segments.length != 4)
+                throw new IllegalArgumentException("Invalid IPv4 address: " + networkInterface());
+
+            try {
+                for (int s = 0; s < segments.length; ++s) {
+                    String segment = segments[s];
+
+                    if (DIGIT_PATTERN.matcher(segment).matches()) {
+                        int seg = Integer.parseInt(segment);
+                        if (seg < 0 || seg > IPV4_MAX_OCTET_VALUE)
+                            throw new IllegalArgumentException("Invalid IPv4 address: " + networkInterface());
+
+                        segmentPred[s] = addr -> segment.equals(addr);
+                    }
+                    else if (RANGE_PATTERN.matcher(segment).matches()) {
+                        final String[] range = segment.split("-");
+                        if (range.length != 2)
+                            throw new IllegalArgumentException("Invalid IPv4 address: " + networkInterface());
```

Comment:
But it's not an IPv4 address, it's some generalized pattern of such address

Reply:
Makes sense to me. Changed to `Invalid IPv4 address pattern: `